### PR TITLE
Populate object store from NOAA obspack

### PR DESCRIPTION
**Pull request by rt17603**
Adding functionality to add data from NOAA ObsPack to an object store from a folder.

Includes **update to metadata checking** - this is now based on a whitelist of selected attributes rather than a blacklist. This is due to the JSON Serializable issue where unnecessary attributes were being added to the metadata and were unable to be stored in JSON format.

At the moment this only able to handle "surface" data but can select this when passed a whole obspack folder. This should be updated when we are able to process data from other sources including "tower" (should be able to incoporate into ObsSurface but need to take account of different heights), "aircraft" and "shipboard" (both ObsMobile).

### CODE REVIEW

File: openghg/store/_populate.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+        param = _param_from_filename(filepath)
+        site = param["site"]
+        project = param["project"]
+        measurement_type = param["measurement_type"]
+
+        if project in projects_to_read:
+            processed = ObsSurface.read_file(filepath, site=site, measurement_type=measurement_type, network="NOAA", data_type="NOAA")
+        elif project in project_names_not_implemented:
+            print(f"Not processing {filepath.name} - no standardisation for {project} data implemented yet.")
+            processed = {}
```
Comment (gareth-j) [inline openghg/store/_populate.py:200]: I get an `unbound variable` issue raised here, will project **always** be in one of these two? Should we have an

```
else:
    processed = {}
```

**Review by gareth-j — CHANGES_REQUESTED** [review]
Looks good, just a few small changes maybe.

File: openghg/standardise/surface/_noaa.py [hunk 2]
```diff
@@ -60,7 +57,7 @@ def parse_noaa(
 def _read_obspack(
     data_filepath: Union[str, Path],
     site: str,
-    inlet: str,
+    inlet: Union[str, None],
```
[the remark this answers is not in the document]
Comment (rt17603) [reply openghg/standardise/surface/_noaa.py:60]: Sure I can update that (same below).

File: openghg/standardise/surface/_noaa.py [hunk 3]
```diff
@@ -120,6 +117,17 @@ def _read_obspack(
     species = clean_string(obspack_ds.attrs["dataset_parameter"])
     network = "NOAA"
 
+    # If inlet is not defined try and derive this from the attribute data
+    if inlet is None:
+        if "dataset_intake_ht" in orig_attrs:
+            # Note: this attribute will be a float stored as a string e.g. 40.0, may want to update to int
```
[the remark this answers is not in the document]
Comment (rt17603) [reply openghg/standardise/surface/_noaa.py:123]: So this is a good question and I'm not sure what the best answer is. I noticed when looking at the heights for some of the other NOAA inlets that there were some fractional heights e.g. 2.5 

So I'm not sure if we do want to insist on this being an integer if we want to allow these sorts of heights to be included.

File: openghg/store/_populate.py (hunk 1 above)
Comment (rt17603) [reply openghg/store/_populate.py:200]: Well project was derived from our defined `project_options` and `project_options_not_yet_implemented` variables which *should* cover all the possibilities we can process.

However, there could be a case for a value not matching, so I can add this in.

File: openghg/store/_populate.py [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+        projects_to_read = project_names
+        print(f"Reading subset of data from {','.join(projects_to_read)}")
+
+    if isinstance(data_directory, str):
+        data_directory = Path(data_directory)
+
+    # ObsPack may contain nc or txt files
+    # Try and extract one and then the other if possible
+    files = _find_noaa_files(data_directory, ".nc")
+    if len(files) == 0:
```
[the remark this answers is not in the document]
Comment (rt17603) [reply openghg/store/_populate.py:185]: I do think `if len(files) == 0` communicates more - i.e. we're expecting back something with extent that we expect to contain values. But I understand that `if not files` is the convention, so will change it if needs be.

File: openghg/standardise/surface/_noaa.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (gareth-j) [reply openghg/standardise/surface/_noaa.py]: Ah OK, maybe it's best to keep this as it is for now then, we don't really have any code that expects an `int` anyway.

**Review by gareth-j — APPROVED** [review]

**Pull request merged**